src/mgmt/srv.c

```diff
@@ -72,314 +72,430 @@ static struct daos_rpc_handler mgmt_handlers[] = {
 #undef X
 
 static void
-process_killrank_request(Drpc__Call *drpc_req, Mgmt__DaosResp *daos_resp)
+pack_daos_response(Mgmt__DaosResp *daos_resp, Drpc__Response *drpc_resp)
 {
-	Mgmt__DaosRank	*pb_rank = NULL;
+	uint8_t	*body;
+	size_t	len;
 
-	/* Response status is populated with SUCCESS on init. */
-	mgmt__daos_resp__init(daos_resp);
+	len = mgmt__daos_resp__get_packed_size(daos_resp);
+	D_ALLOC(body, len);
+	if (body == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
+		D_ERROR("Failed to allocate drpc response body\n");
+		return;
+	}
 
-	/* Unpack the daos request from the drpc call body */
-	pb_rank = mgmt__daos_rank__unpack(
-		NULL, drpc_req->body.len, drpc_req->body.data);
+	if (mgmt__daos_resp__pack(daos_resp, body) != len) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
+		D_ERROR("Unexpected num bytes for daos resp\n");
+		return;
+	}
 
-	if (pb_rank == NULL) {
-		daos_resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
-		D_ERROR("Failed to extract rank from request\n");
+	/* Populate drpc response body with packed daos response */
+	drpc_resp->body.len = len;
+	drpc_resp->body.data = body;
+}
 
+static void
+process_killrank_request(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	Mgmt__DaosRank		*req = NULL;
+	Mgmt__DaosResp		*resp = NULL;
+
+	/* Unpack the inner request from the drpc call body */
+	req = mgmt__daos_rank__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+	if (req == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
```

Comment:
Don't need to address it now, but maybe we should add a dRPC proto failure status for this type of case, and perhaps a dRPC NOMEM or even a generic system failure for the case where we fail to allocate part of the response.